src/core/image.rs

```diff
@@ -89,6 +83,114 @@ where
     }
 }
 
+#[derive(Debug)]
+pub struct RunnableImage<I: Image> {
+    image: I,
+    image_args: I::Args,
+    image_tag: Option<String>,
+    name: Option<String>,
+    network: Option<String>,
+    env_vars: BTreeMap<String, String>,
+    volumes: BTreeMap<String, String>,
+    ports: Option<Vec<Port>>,
+}
+
+impl<I: Image> RunnableImage<I> {
+    pub fn image(&self) -> &I {
+        &self.image
+    }
+
+    pub fn image_args(&self) -> &I::Args {
+        &self.image_args
+    }
+
+    pub fn network(&self) -> &Option<String> {
+        &self.network
+    }
+
+    pub fn name(&self) -> &Option<String> {
+        &self.name
+    }
+
+    pub fn env_vars(&self) -> &BTreeMap<String, String> {
+        &self.env_vars
+    }
+
+    pub fn volumes(&self) -> &BTreeMap<String, String> {
+        &self.volumes
+    }
+
+    pub fn ports(&self) -> &Option<Vec<Port>> {
+        &self.ports
+    }
+}
```

Comment:
Do we need to expose all of these publicly? I would like the public API to be as small as possible so we have to release less breaking changes.